fdbserver/ClusterController.actor.cpp

```diff
@@ -3095,6 +3095,10 @@ ACTOR Future<Void> workerAvailabilityWatch(WorkerInterface worker,
 				cluster->removedDBInfoEndpoints.insert(worker.updateServerDBInfo.getEndpoint());
 				cluster->id_worker.erase(worker.locality.processId());
 				cluster->updateWorkerList.set(worker.locality.processId(), Optional<ProcessData>());
+				TraceEvent("ClusterControllerWorkerFailed", cluster->id)
+					.detail("ProcessId", worker.locality.processId())
+					.detail("ProcessClass", failedWorkerInfo.details.processClass.toString())
```

Comment:
`failedWorkerInfo` is a reference into `id_worker` which gets erased earlier, probably that's why test is failing.